### PR TITLE
Piecewise simplification (remove XFAIL after corrections)

### PR DESCRIPTION
**Pull request by smichr**
#### References to other Issues or PRs

continue #25623, correcting Piecewise simplification conditions

#### Release Notes

* functions
    * a logical error in Piecewise simplification was corrected

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
This is not something that is changed in this PR but one thing that I just noticed is:
```python
In [1]: p = Piecewise((f(1), (0 < x) & (x < 2)), (f(2), (1 < x) & (x < 3)))

In [2]: p
Out[2]: 
⎧f(1)  for x > 0 ∧ x < 2
⎨                       
⎩f(2)  for x > 1 ∧ x < 3

In [3]: p.simplify()
Out[3]: 
⎧f(1)  for x > 0 ∧ x < 2
⎪                       
⎨f(2)  for x ≥ 2 ∧ x < 3
⎪                       
⎩nan       otherwise  
```
I don't think that adding the redundant nan case explicitly is really a simplification so ideally simplify would not add it.

File: sympy/functions/elementary/piecewise.py [hunk 1]
```diff
@@ -1172,18 +1172,27 @@ def include(c, x, a):
                 cset = iv - covered
                 if not cset:
                     continue
+                try:
+                    a = cset.inf
+                    incl_a = include(c, x, a)
+                except NotImplementedError:
+                    pass # continue with the given `a`
```
Comment (oscarbenjamin) [inline sympy/functions/elementary/piecewise.py:1179]: If this is catching `NotImplementedError` from `cset.inf` then that is the only thing that should be in the try block:
```
                try:
                    a = cset.inf
                except NotImplementedError:
                    pass # continue with the given `a`
                else:
                    incl_a = include(c, x, a)
```

**Comment by oscarbenjamin** [comment]
The logic here is asymmetric:
```python
In [12]: p1 = Piecewise((f(1), (0 <= x) & (x <= 2)), (f(2), (1 <= x) & (x <= 3)))

In [13]: p2 = Piecewise((f(1), (3 <= x) & (x <= 1)), (f(2), (0 <= x) & (x <= 2)))

In [14]: p1
Out[14]: 
⎧f(1)  for x ≥ 0 ∧ x ≤ 2
⎨                       
⎩f(2)  for x ≥ 1 ∧ x ≤ 3

In [15]: p1.simplify()
Out[15]: 
⎧f(1)  for x ≥ 0 ∧ x ≤ 2
⎪                       
⎨f(2)  for x ≤ 3 ∧ x > 2
⎪                       
⎩nan       otherwise    

In [16]: p2
Out[16]: 
⎧f(1)  for x ≥ 3 ∧ x ≤ 1
⎨                       
⎩f(2)  for x ≥ 0 ∧ x ≤ 2

In [17]: p2.simplify()  # not correct
Out[17]: {f(2)  for x ≥ 0 ∧ x ≤ 2
```

**Comment by smichr** [comment]
> The logic here is asymmetric:

The problem is with `_intervals`. I will investigate.

File: sympy/functions/elementary/piecewise.py (hunk 1 above)
Comment (smichr) [reply sympy/functions/elementary/piecewise.py:1179]: Thanks for the reminder of the `else`-after block.

Comment (smichr) [inline sympy/functions/elementary/piecewise.py:1179]:
```suggestion
                except NotImplementedError:
                    pass # continue with the given `a`
                else:
                    incl_a = include(c, x, a)
```

**Comment by smichr** [comment]
I will open an issue for the two points you raised and handle that as a separate PR.

**Comment by smichr** [comment]
> ```python
> p2.simplify()  # not correct
> ```

It looks ok to me -- the first clause of the Piecewise is empty since there is no `x` that is greater than 3 and less than 1.

**Pull request merged**